packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -352,48 +352,74 @@ typedef bool RoutePredicate(Route<dynamic> route);
 /// The app's home page route is named '/' by default and other routes are
 /// given pathnames by convention.
 ///
-/// ### Defining a popup route
+/// ### Routes can return a value
+///
+/// When a route is pushed to ask the user for a value, the value can be
+/// returned via the [pop] method's result parameter.
+///
+/// Methods that push a route return a Future. The Future resolves when
+/// the route is popped and the Future's value is the [pop] method's result
+/// parameter.
+///
+/// For example if we wanted to ask the to press 'OK' to confirm an
+/// operation we could await the result of [Navigator.push]:
+///
+/// ```
+/// bool value = await Navigator.of(context).push(new MaterialPageRoute<bool>(
+///   builder: (BuildContext context) {
+///     return new Center(
+///       child: new FlatButton(
```

Comment:
Scaffold builds a Material around `body`.  In this example, there is no scaffold, which means the code will assert.